temporalio/workflow.py

```diff
@@ -1099,18 +1104,28 @@ def __init__(
         super().__init__(logger, extra or {})
         self.workflow_info_on_message = True
         self.workflow_info_on_extra = True
+        self.full_workflow_info_on_extra = False
         self.log_during_replay = False
 
     def process(
         self, msg: Any, kwargs: MutableMapping[str, Any]
     ) -> Tuple[Any, MutableMapping[str, Any]]:
         """Override to add workflow details."""
-        if self.workflow_info_on_message or self.workflow_info_on_extra:
+        if (
+            self.workflow_info_on_message
+            or self.workflow_info_on_extra
+            or self.full_workflow_info_on_extra
+        ):
             runtime = _Runtime.maybe_current()
             if runtime:
                 if self.workflow_info_on_message:
                     msg = f"{msg} ({runtime.logger_details})"
                 if self.workflow_info_on_extra:
+                    # Extra can be absent or None, this handles both
+                    extra = kwargs.get("extra", None) or {}
+                    extra["temporal_workflow"] = runtime.logger_details
+                    kwargs["extra"] = extra
+                if self.full_workflow_info_on_extra:
```

Comment:
Can you verify we have test coverage for this option (and the activity full info) just in case?

Reply:
Ok

Reply:
Added assertion for this field being present (no assertion for the inverse, don't think it's worth a new test for all options permutations here).